aten/src/ATen/native/transformers/cuda/sdp_utils.h

```diff
@@ -29,6 +30,46 @@ struct sdp_params {
   bool is_causal;
 };
 
+inline std::array<SDPBackend, num_backends> priority_order(sdp_params params) {
```

Comment:
Another way to implement this (and I think it's kind of what this is), is to modify `use_flash_attention` and `use_mem_efficient_attention` to return an integer or to create new functions that return integers.

These integers then are the estimated number of operations performed by the respective fused kernel. This is similar to [`estimate_matmul_time`](https://github.com/pytorch/pytorch/blob/master/torch/_inductor/triton_ops/mm_perf_model.py#L4).

You then pick the one that returns the lowest number of operations. And if the number of operations is negative, well then the kernel doesn't apply.